plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -45,40 +45,14 @@ protected void setup(Binder binder)
-            newOptionalBinder(binder, Key.get(new TypeLiteral<ExecutorService>() {}, ThriftHiveWriteStatisticsExecutor.class))
-                    .setDefault().toInstance(newFixedThreadPool(1, threadsNamed("http-thrift-statistics-write-%s")));
-            ThriftHttpMetastoreConfig httpMetastoreConfig = buildConfigObject(ThriftHttpMetastoreConfig.class);
-            boolean hasHttpsMetastore = staticMetastoreConfig.getMetastoreUris().stream().anyMatch(uri -> "https".equalsIgnoreCase(uri.getScheme()));
-            if (hasHttpsMetastore && httpMetastoreConfig.getHttpBearerToken().isEmpty()) {
-                throw new IllegalStateException("'hive.metastore.http.client.bearer-token' must be set while using https metastore URIs in 'hive.metastore.uri'");
-            }
-
-            if (!hasHttpsMetastore && httpMetastoreConfig.getHttpBearerToken().isPresent()) {
-                throw new IllegalStateException("'hive.metastore.http.client.bearer-token' must not be set while using http metastore URIs in 'hive.metastore.uri'");
-            }
-            if (httpMetastoreConfig.getAuthenticationMode().isEmpty()) {
-                throw new IllegalStateException("'hive.metastore.http.client.authentication.type' must be set while using http/https metastore URIs in 'hive.metastore.uri'");
-            }
-        }
-        else {
-            OptionalBinder.newOptionalBinder(binder, ThriftMetastoreClientFactory.class)
-                    .setDefault().to(DefaultThriftMetastoreClientFactory.class).in(Scopes.SINGLETON);
-            binder.bind(TokenAwareMetastoreClientFactory.class).to(StaticTokenAwareMetastoreClientFactory.class).in(Scopes.SINGLETON);
-            configBinder(binder).bindConfig(ThriftMetastoreConfig.class);
-            newOptionalBinder(binder, Key.get(new TypeLiteral<ExecutorService>() {}, ThriftHiveWriteStatisticsExecutor.class))
-                    .setDefault().toProvider(ThriftHiveMetastoreStatisticExecutorProvider.class).in(Scopes.SINGLETON);
-            install(new ThriftMetastoreAuthenticationModule());
-            binder.bind(ThriftMetastoreFactory.class).to(ThriftHiveMetastoreFactory.class).in(Scopes.SINGLETON);
-        }
-
+        OptionalBinder.newOptionalBinder(binder, ThriftMetastoreClientFactory.class)
+                .setDefault().to(DefaultThriftMetastoreClientFactory.class).in(Scopes.SINGLETON);
+        binder.bind(TokenAwareMetastoreClientFactory.class).to(StaticTokenAwareMetastoreClientFactory.class).in(Scopes.SINGLETON);
+        configBinder(binder).bindConfig(ThriftMetastoreConfig.class);
+        newOptionalBinder(binder, Key.get(new TypeLiteral<ExecutorService>() {}, ThriftHiveWriteStatisticsExecutor.class))
+                .setDefault().toProvider(ThriftHiveMetastoreStatisticExecutorProvider.class).in(Scopes.SINGLETON);
+        install(new ThriftMetastoreAuthenticationModule());
+        binder.bind(ThriftMetastoreFactory.class).to(ThriftHiveMetastoreFactory.class).in(Scopes.SINGLETON);
```

Comment:
Not quite sure whether the following changes from https://github.com/trinodb/trino/pull/20371/  still make sense:

https://github.com/trinodb/trino/blob/d872e37cd66ed97457540c61475ec2798e94f448/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java#L375-L377